src/resolve/resolved-bus.c

```diff
@@ -1580,6 +1582,132 @@ static int bus_method_reset_server_features(sd_bus_message *message, void *userd
         return sd_bus_reply_method_return(message, NULL);
 }
 
+static int bus_method_register_service(sd_bus_message *message, void *userdata, sd_bus_error *error) {
+        _cleanup_(dnssd_service_freep) DnssdService *service = NULL;
+        _cleanup_free_ char *path = NULL;
+        _cleanup_free_ char *instance_name = NULL;
+        Manager *m = userdata;
+        DnssdService *s = NULL;
+        DnsTxtItem *last = NULL;
+        int r;
+
+        assert(message);
+        assert(m);
+
+        if (m->mdns_support != RESOLVE_SUPPORT_YES)
+                return sd_bus_error_set_errno(error, EPROTONOSUPPORT);
+
+        service = new0(DnssdService, 1);
+        if (!service)
+                return sd_bus_error_set_errno(error, ENOMEM);
```

Comment:
use sd_bus_error_setf(SD_BUS_ERROR_MEMORY, …) here instead...

sd_bus_error_set_errno() calls should only be used if we receieved an errno from somebody else we want to propagate. We shouldn't use it when we generate an error ourselves. (similar above and elswehere)